src/opts.rs

```diff
@@ -0,0 +1,83 @@
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+pub const DEFAULT_MANIFEST_FILE: &str = "spin.toml";
+pub const APP_CONFIG_FILE_OPT: &str = "APP_CONFIG_FILE";
+pub const ADDRESS_OPT: &str = "ADDRESS";
+pub const BINDLE_ID_OPT: &str = "BINDLE_ID";
+pub const BINDLE_SERVER_URL_OPT: &str = "BINDLE_SERVER_URL";
+pub const BINDLE_URL_ENV: &str = "BINDLE_URL";
+pub const APP_LOG_DIR: &str = "APP_LOG_DIR";
+pub const TLS_CERT_FILE_OPT: &str = "TLS_CERT_FILE";
+pub const TLS_KEY_FILE_OPT: &str = "TLS_KEY_FILE";
+pub const TLS_CERT_ENV_VAR: &str = "SPIN_TLS_CERT";
+pub const TLS_KEY_ENV_VAR: &str = "SPIN_TLS_KEY";
+pub const BUILDINFO_OPT: &str = "BUILDINFO";
+pub const INSECURE_OPT: &str = "INSECURE";
+pub const STAGING_DIR_OPT: &str = "STAGING_DIR";
+pub const HIPPO_SERVER_URL_OPT: &str = "HIPPO_SERVER_URL";
+pub const HIPPO_URL_ENV: &str = "HIPPO_URL";
+pub const JSON_MIME_TYPE: &str = "application/json";
+pub const BUILD_UP_OPT: &str = "UP";
+
+#[derive(StructOpt, Debug)]
+pub struct UpOpts {
```

Comment:
I would lean towards leaving this in `up.rs`. You're importing `UpCommand` in `build` anyway, and splitting these into another file means more jumping between files when making changes to `up`.